### PR TITLE
Feat/Upload multiple files in different fields

### LINKED ISSUE #1493: Upload multiple files with different field names
## Sorting

- **I'm submitting a ...**

  - [X] bug report
  - [ ] feature request
  - [ ] support request

- I confirm that I
  - [X] used the [search](https://github.com/lukeautry/tsoa/search?type=Issues) to make sure that a similar issue hasn't already been submit

## Expected Behavior

I want to upload multiple files with different field names but this is not possible. I can only upload multiple files with the same field name.

## Current Behavior

```ts

    @Patch("/uploadFile")
    @Security("jwt", ["user"])
    @Middlewares(checkMiddleware.checkUserIsEnabled)
    @SuccessResponse("200", "Files uploaded successfully")
    public async uploadFile(
        @UploadedFile("file1") profileImage: Express.Multer.File,
        @UploadedFile("file2") profileImage1: Express.Multer.File,
        @Request() expressRequest: express.Request
    ): Promise<Obj> {
        this.setStatus(HttpStatusCode.OK);
        const req = expressRequest as RequestCustomUser;
        const obj = await FileService.uploadFile(
            req.token,
            file1,
            file2,
        );
        return obj;
    }
```

## Context (Environment)

Version of the library: 5.1.1
Version of NodeJS: 20

- Confirm you were using yarn not npm: [X]

### PR DESCRIPTION
**Pull request by jackey8616**
### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/lukeautry/tsoa/tree/master/docs/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/lukeautry/tsoa/pulls) for the same update/change?
- [x] Have you written unit tests?
- [x] Have you written unit tests that cover the negative cases (i.e.: if bad data is submitted, does the library respond properly)?
- [x] This PR is associated with an existing issue?

**Closing issues**
Closes #1493 
Put `closes #XXXX` (where XXXX is the issue number) in your comment to auto-close the issue that your PR fixes.

### If this is a new feature submission:

- [x] Has the issue had a maintainer respond to the issue and clarify that the feature is something that aligns with the [goals](https://github.com/lukeautry/tsoa#goal) and [philosophy](https://github.com/lukeautry/tsoa#philosophy) of the project?

**Potential Problems With The Approach**
n/a

**Test plan**
Tested with unit test with multiple decorator usage.

### CODE REVIEW

File: packages/cli/src/routeGeneration/templates/express.hbs [hunk 1]
```diff
@@ -66,7 +66,7 @@ export function RegisterRoutes(app: Router) {
             authenticateMiddleware({{json security}}),
             {{/if}}
             {{#if uploadFile}}
-            upload.single('{{uploadFileName}}'),
+            upload.fields({{json uploadFileName}}),
```
Comment (WoH) [inline packages/cli/src/routeGeneration/templates/express.hbs:69]: This will not break single files?

**Review by WoH — COMMENTED** [review]
Just a question about existing coverage

File: packages/cli/src/routeGeneration/templates/express.hbs (hunk 1 above)
Comment (jackey8616) [reply packages/cli/src/routeGeneration/templates/express.hbs:69]: I also rewrite the uploadFileName field in routeGenerator.tsL105
Check [here](https://github.com/lukeautry/tsoa/pull/1541/files#diff-fa3f83789f7f648679cd056a5564dead51915ecc3903778f9808122e50be44eeR105-R108).

All exists tests are not been modified.

**Comment by jackey8616** [comment]
@WoH This PR is also good to go. thx

**Comment by WoH** [comment]
Can you please rebase?

**Comment by jackey8616** [comment]
> Can you please rebase?

Rebase completed.

**Comment by yyoshiki41** [comment]
Thank you for your ongoing development efforts!
Could you inform me when the npm package will be released?

**Pull request merged**